ui/src/shared/apis/query.ts

```diff
@@ -138,8 +139,9 @@ const processErrorResponse = async (
     const body = await response.text()
     const json = JSON.parse(body)
     const message = json.message || json.error
+    const code = json.code || ''
```

Comment:
should this fall back to `null` or `undefined` rather than `''`? trying to think of the meanings of all these things:

`undefined`: doesn't exist. the same as doing `const code = json.code;`
`null`: has been assigned a value of: `no value`
`''`: has been assigned a value of: `no value`, represented as a string

the api, if it returns anything, will return a string. if it doesn't return anything, the value of `code` will be `undefined`. an empty string _might_ imply that we got a response back from the API